site/app/models/gradeable/Gradeable.php

```diff
@@ -0,0 +1,446 @@
+    protected $type = GradeableType::ELECTRONIC_FILE;
+    /** @property @var bool If the gradeable should be graded per registration section (true) or rotating sections(false) */
+    protected $grade_by_registration = true;
+    /** @property @var \DateTime The so-called 'TA Beta-Testing' date.  This is when the gradeable appears for TA's */
+    protected $ta_view_start_date = null;
+    /** @property @var \DateTime The date that graders may start grading */
+    protected $grade_start_date = null;
+    /** @property @var \DateTime The date that grades will be released to students */
+    protected $grade_released_date = null;
+    /** @property @var \DateTime The date after which only instructors may change grades (aka when grades are 'due') */
+    protected $grade_locked_date = null;
+    /** @property @var int The minimum user group that can grade this gradeable (1=instructor) */
+    protected $min_grading_group = 1;
+    /** @property @var string The syllabus classification of this gradeable */
+    protected $syllabus_bucket = "Homework";
+
+    /** @property @var Component[] An array of all of this gradeable's components */
+    protected $components = array();
+
+    /* Properties exclusive to numeric-text/checkpoint gradeables */
+
+    /** @property @var string The overall ta instructions for grading (numeric-text/checkpoint only) */
+    protected $ta_instructions = "";
+
+    /* Properties exclusive to electronic gradeables */
+
+    /** @property @var string The location of the autograding configuration file */
+    protected $autograding_config_path = "";
+    /** @property @var string[] The object that contains the autograding config data */
+    private $autograding_config = null;
+    /** @property @var bool If the gradeable is using vcs upload (true) or manual upload (false) */
+    protected $vcs = false;
+    /** @property @var string The subdirectory within the VCS repository for this gradeable */
+    protected $vcs_subdirectory = "";
+    /** @property @var bool If the gradeable is a team assignment */
+    protected $team_assignment = false;
+    /** @property @var int The maximum team size (if the gradeable is a team assignment) */
+    protected $team_size_max = 0;
+    /** @property @var \DateTime The deadline for joining teams (if the gradeable is a team assignment) */
+    protected $team_lock_date = null;
+    /** @property @var bool If the gradeable is using any manual grading */
+    protected $ta_grading = false;
+    /** @property @var bool If students can view submissions */
+    protected $student_view = false;
+    /** @property @var bool If students can make submissions */
+    protected $student_submit = false;
+    /** @property @var bool If students can download submitted files */
+    protected $student_download = false;
+    /** @property @var bool If students can view/download any version of the submitted files, or just the active version */
+    protected $student_any_version = false;
```

Comment:
Variable name (while probably matching the db) is not very intuitive without having to read the doc comment. May want to use `$student_download_any_version` to inform people this is about _downloading_